snippets/survival_analysis/KaplanMeier_curves/KaplanMeier.Rmd

```diff
@@ -1,83 +1,101 @@
+  "death" = disease unrelated death  
+  "alive" = alive at given survival time  
+  "alive w metastases" = alive with metasatses at given survival time  
+
+# Prepare data for Kaplan Meier survival curves
+
+Survival data is censored when the event has not yet occurred for a patient
+at a given time or the patient is lost to follow-up during the study. In
+Kaplan Meier survival curves censored patients are indicated by a cross
+on the curve.
+A more detailed explanation can be found here: 
+http://www.sthda.com/english/wiki/survival-analysis-basics#censoring
 
-#Censoring for overall survival (1 = event (death due to disease), 0 = censored)
+```{r}
+# Censoring for overall survival (1 = event (death due to disease), 0 = censored)
 surv_dat$censoringOS = 0
 surv_dat$censoringOS[surv_dat$Patientstatus == 'death by primary disease'] = 1
 
-#Censoring for disease free survival (1 = event (recurrence), 0 = censored)
+# Censoring for disease free survival (1 = event (recurrence), 0 = censored)
 surv_dat$censoringDFS = 0
 surv_dat$censoringDFS[surv_dat$Patientstatus == 'alive w metastases'] = 1
 surv_dat$censoringDFS[surv_dat$OSmonth > surv_dat$DFSmonth] = 1 
 
 
-#Survival object for OS and DFS
+# Build the standard survival object with the Surv function from the package
+# "survival". A plus behind the survival time inicates censoring in the survival object.
 SurvObj_OS <- Surv(surv_dat$OSmonth, surv_dat$censoringOS)
 SurvObj_DFS <- Surv(surv_dat$DFSmonth, surv_dat$censoringDFS)
 
 ```
 
-# Compute OS survival curves for the different pathology grades (for DFS just use SurvObj_DFS instead)
-```{r}
+# Compute Kaplan Meier estimates
+
+Use the survfit function from the "survival" package in order to produce
+the Kaplan Meier estimates of probability of survival over time.
 
-#Compute survival curve across all patients
+```{r}
+# Compute the survival curve across all patients in the survival object.
+# Here we use overall suvrival. For disease free just replace the SurvObj_OS
+# with SurvObj_DFS.
 KM_all <- survfit(SurvObj_OS ~ 1)
 
-#Compute survival curves for each grade
+# Compute survival curves for different groups of patients (here grade).
 surv_dat$grade = as.factor(surv_dat$grade)
 KM_groups <- survfit(SurvObj_OS ~ surv_dat$grade)
 
 ```
 
-# Plot KM survival curves overlaid in one panel
-```{r}
-library(RColorBrewer)
+# Plot survival curves overlaid in one panel
+
+Use the plot function from the survival package to visualize the KM
+survival curves. Alternatively, one could use the "ggfortify" package
+providing the ggplot2::autoplot function.
 
-#Color map
+```{r}
+# Color map for plotting
 cols = brewer.pal(3,"Dark2")
 
-#All patinets survival curve: black, grade survival curves: color (CIs for strata not plotted)
+# All patinets survival curve: black, grade survival curves: color (CIs for strata not plotted)
 plot(KM_groups,mark.time = T,col = cols[as.numeric(levels(surv_dat$grade))],conf.int = F)
```

Comment:
Just out of curiosity: would there be a nicer `ggplot2` function to do this job?

Reply:
The `broom` example for `survfit` has a nice example how to ggplot2 this:
https://rdrr.io/cran/broom/man/tidy.survfit.html

Reply:
I didn't know the broom package but it look s convenient!